experimental/trusted_shuffler/client/src/main.rs

```diff
@@ -32,27 +35,106 @@ pub struct Opt {
         default_value = "http://localhost:8080"
     )]
     server_url: String,
+    #[structopt(long, help = "The QPS we are aiming to simulate", default_value = "10")]
+    qps: u32,
+    #[structopt(
+        long,
+        help = "How many rounds, i.e., seconds, we sent requests",
+        default_value = "1"
+    )]
+    rounds: u32,
 }
 
-#[tokio::main]
+#[tokio::main(flavor = "multi_thread", worker_threads = 2)]
 async fn main() -> anyhow::Result<()> {
-    env_logger::init();
+    env_logger::builder()
+        .format_timestamp(None)
+        .format_level(false)
+        .format_module_path(false)
+        .format_target(false)
+        .init();
     let opt = Opt::parse();
 
-    let request = "Request 1";
-    info!("Sending request: {}", request);
-    let url = format!("{}/request", &opt.server_url);
-
-    let request_start = Instant::now();
-    let response = send_request(&url, Method::POST, request.as_bytes())
-        .await
-        .context("Couldn't receive response")?;
-    let request_duration = request_start.elapsed();
-
-    let parsed_response = String::from_utf8(response).context("Couldn't decode response body")?;
-    info!(
-        "Received response: {} in {:?}",
-        parsed_response, request_duration
-    );
+    let expected_qps = opt.qps;
+    let rounds = opt.rounds;
+
+    // Start with giving one second iterations.
+    let start_time = Instant::now();
+
+    // Header of table printed by logger.
+    log::info!("phase,id,request_elapsed_in_ms,response_elapsed_in_ms");
+
+    let mut clients = vec![];
+
+    for (n, p) in Sieve::new()
+        .iter()
+        .take((expected_qps * rounds) as usize)
+        .enumerate()
+    {
+        let server_url = opt.server_url.clone();
+
+        // We currently spawn a new client for every new request. Alternatively we could re-use the
+        // client for every round.
+        clients.push(tokio::spawn(async move {
+            let request = format!("{}", p);
+            let request_start = start_time.elapsed();
+
+            let url = format!("{}/request", server_url);
+            let response = send_request(&url, Method::POST, request.as_bytes())
+                .await
+                .context("Couldn't receive response");
+
+            let request_duration = start_time.elapsed();
+
+            if let Ok(response) = response {
+                // We don't check whether response is exepected response.
```

Comment:
Out of curiosity, why not? It seems you already implemented all the fancy logic with primes and stuff, just to ignore the response? :)

Reply:
I was just curious why there is logic to generate some request payload, but it then does not get used for anything. It seems that if we assume that the server is just doing echos, the logic here should be able to know exactly what to expect back and check that.